[DPE-9311] Initial user documentation (#24)
canonical/valkey-operator: docs/how-to/scale-horizontally.md

# How to scale horizontally

Adding and removing units from a Valkey deployment is done by scaling [Juju units](https://juju.is/docs/juju/unit). 
 
## Add a unit

> You can add additional units to your deployed Valkey application with the following command:

Reviewer: Chore: Empty line around headings.

Suggestion: To make this how-to guide more imperative, let's replace descriptive phrases with "can" by more actionable format: 

`To do something:`
`<command>`

```suggestion

To add additional units:
```

Please apply to all how-to guides where possible.

Author: updated

# How to scale horizontally

> Adding and removing units from a Valkey deployment is done by scaling [Juju units](https://juju.is/docs/juju/unit). 

Reviewer: Suggestion: Let's adjust wording to avoid repeating units (adding units by scaling units.

```suggestion
Horizontal scaling of a Valkey deployment is done by adding and removing [Juju units](https://juju.is/docs/juju/unit). 
```

 
## Add a unit

To add additional units to your deployed Valkey application, run the following command:

```shell

> juju add-unit valkey -n 1

Reviewer: Question: We do the VM way here. Do we need to provide a [K8s-style](https://documentation.ubuntu.com/juju/3.6/reference/juju-cli/list-of-juju-cli-commands/scale-application/) scaling command too?

Author: This command works on K8s, too. I've ensured the commands in the docs work on both substrates throughout, except explicitly stated otherwise.

```text
Model     Controller      Cloud/Region        Version  SLA          Timestamp
tutorial  k8s-controller  microk8s/localhost  3.6.14   unsupported  19:21:12+01:00

App                       Version  Status  Scale  Charm                     Channel   Rev  Address         Exposed  Message
self-signed-certificates           active      1  self-signed-certificates  1/stable  586  10.152.183.111  no       
valkey                             active      4  valkey                    9/edge     11  10.152.183.123  no       

Unit                         Workload  Agent  Address      Ports  Message
self-signed-certificates/0*  active    idle   10.1.44.89          
valkey/0*                    active    idle   10.1.44.126         
valkey/1                     active    idle   10.1.44.117         
valkey/2                     active    idle   10.1.44.127         
valkey/3                     active    idle   10.1.44.68          
```

### Remove units

Removing a unit from the application scales down the replicas. If you currently have
three units, one is the primary and two are replicas. Removing a unit will reduce the

> number of replicas to one.

Reviewer: Suggestion: Let's match the description to the example below and improve wording to avoid user confusion.

```suggestion
Removing a unit from the application scales down the application. If you currently have
four units, then only one is the primary and three are replicas. Removing a unit will reduce the
number of replicas to two and the total number of units to three.
```

Before scaling down, list all the units with `juju status`:

* `valkey/0`
* `valkey/1`
* `valkey/2`
* `valkey/3`

To scale the application down to three units, run:

```shell
juju remove-unit valkey --num-units 1
```

Safely removing the unit will take a few moments. You’ll know that the unit was

> successfully removed when `juju status` reports:

Reviewer: Suggestion: Wording improvements for clarity:

```suggestion
successfully removed when `juju status` reports all units `active`/`idle` again:
```
